File: Backend/marketplace_server/entity/subscription.py
from sqlalchemy import Column, String, BigInteger, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from db import Base

class Subscription(Base):
    __tablename__ = 'subscriptions'
    id = Column(UUID(as_uuid=True), primary_key=True)
    user_id = Column(BigInteger, ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
    status = Column(String, nullable=False)
    plan = Column(String, nullable=False, default='pro')
    provider = Column(String)
    external_id = Column(String)
    started_at = Column(DateTime)
    renews_at = Column(DateTime)
    canceled_at = Column(DateTime)

File: serverless/rust/entity/subscription.rs
use sea_orm::entity::prelude::*;
use sqlx::FromRow;
use uuid::Uuid;
use chrono::{NaiveDateTime, DateTime};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, FromRow)]
#[sea_orm(table_name = "subscriptions")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: Uuid,
    pub user_id: i64,
    pub status: String,
    pub plan: String,
    pub provider: Option<String>,
    pub external_id: Option<String>,
    pub started_at: Option<DateTime>,
    pub renews_at: Option<DateTime>,
    pub canceled_at: Option<DateTime>,
}

#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}